packages/expo-calendar/android/src/main/java/expo/modules/calendar/CalendarEvenFactory.kt

```diff
@@ -0,0 +1,80 @@
+package expo.modules.calendar
+
+import android.content.ContentValues
+import org.unimodules.core.arguments.ReadableArguments
+import java.util.*
+
+class CalendarEvenFactory(
+  private val eventDetails: ReadableArguments
+) {
+  val eventValues = ContentValues()
+
+  fun put(key: String, value: String) {
+    eventValues.put(key, value)
+  }
+
+  fun put(key: String, value: Int) {
+    eventValues.put(key, value)
+  }
+
+  fun put(key: String, value: Long) {
+    eventValues.put(key, value)
+  }
+
+  fun put(key: String, value: Boolean) {
+    eventValues.put(key, value)
+  }
+
+  fun putNull(key: String) {
+    eventValues.putNull(key)
+  }
+
+  fun putEventStrings(vararg pairs: Pair<String, String>): CalendarEvenFactory {
+    pairs.forEach { putEventString(it.first, it.second) }
+    return this
+  }
```

Comment:
```suggestion
  fun putEventStrings(vararg pairs: Pair<String, String>) = apply {
    pairs.forEach { putEventString(it.first, it.second) }
  }
```
A better way to creat a builder in Kotlin ;)